pkg/receive/config.go

```diff
@@ -29,6 +29,28 @@ var (
 	errEmptyConfigurationFile = errors.New("configuration file is empty")
 )
 
+type ReceiverMode string
+
+const (
+	RouterOnly     ReceiverMode = "RouterOnly"
+	IngestorOnly   ReceiverMode = "IngestorOnly"
+	RouterIngestor ReceiverMode = "RouterIngestor"
+)
+
+// DetermineMode returns the ReceiverMode that this receiver is configured to run in.
+// This is used to configure this Receiver's forwarding and ingesting behavior at runtime.
+func DetermineMode(hashringSpecified, localEndpointSpecified bool) ReceiverMode {
+	switch {
+	case hashringSpecified && localEndpointSpecified:
+		return RouterIngestor
+	case hashringSpecified && !localEndpointSpecified:
```

Comment:
Let's add note here that IF in this case hashring specifies the same router (aciddently or not) it will cause crash loop (fork bomb) - this is fine, but let's just explicitly write it to not forget